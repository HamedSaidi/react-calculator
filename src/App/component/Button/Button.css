.component-button {
  cursor: pointer;
  width: 100%;
  height: 100px;
  text-align: center;
  background-color: #fff;
  border: 0.125px solid #909090;
  font-size: 1.5rem;
  margin: 0 1px 0 0;
}

.equalSign {
  height: 200px;
}

.lightBlue {
  background-color: #98e2fb;
}

.red {
  background-color: #fa6464;
}

.component-button:hover {
  letter-spacing: 1px;
  background-color: #f2f2f2;
  transition: all 0.4s ease 0s;
}

.lightBlue:hover {
  background-color: #4ecdf8;
}

.red:hover {
  background-color: #f24444;
}

.component-button:last-child {
  margin-right: 0;
}
