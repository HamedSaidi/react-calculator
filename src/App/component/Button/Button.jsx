import React from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';

import {
    EQUAL, PLUS, MINUS, AC,
} from '../../constants';
import './Button.css';

export default class Button extends React.Component {
  static propTypes = {
      name: PropTypes.string.isRequired,
      clickHandler: PropTypes.func.isRequired,
  }

  handleClick = () => {
      const { name, clickHandler } = this.props;

      clickHandler(name);
  }

  render() {
      const { name } = this.props;
      const classes = cx('component-button', {
          equalSign: name === EQUAL,
          lightBlue: [PLUS, MINUS, EQUAL].includes(name),
          red: name === AC,
      });

      return (
          <button className={classes} type="submit" onClick={this.handleClick}>
              {name}
          </button>
      );
  }
}
