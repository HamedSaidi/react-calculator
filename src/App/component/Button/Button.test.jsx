import React from 'react';
import { shallow } from 'enzyme';

import {
    EQUAL, PLUS, MINUS, AC, NUMBERS,
} from '../../constants';
import Button from '.';

describe('Button', () => {
    let consoleSpy;

    beforeAll(() => {
        consoleSpy = jest.spyOn(console, 'error');
    });

    it('renders the Button component and matches snapshot', () => {
        const component = shallow(<Button name="test" clickHandler={jest.fn()} />);

        expect(component).toMatchSnapshot();
    });

    it('renders the PLUS and MINUS button with lightBlue class', () => {
        [PLUS, MINUS].forEach((name) => {
            const component = shallow(<Button name={name} clickHandler={jest.fn()} />);

            expect(component.find('.lightBlue')).toHaveLength(1);
            expect(component.find('.equalSign')).toHaveLength(0);
        });
    });

    it('renders numbers buttons without lightBlue and equalSign classes', () => {
        NUMBERS.split('').forEach((name) => {
            const component = shallow(<Button name={name} clickHandler={jest.fn()} />);

            expect(component.find('.lightBlue')).toHaveLength(0);
            expect(component.find('.equalSign')).toHaveLength(0);
        });
    });

    it('renders the EQUAL button with lightBlue and equalSign classes', () => {
        const component = shallow(<Button name={EQUAL} clickHandler={jest.fn()} />);

        expect(component.find('.lightBlue')).toHaveLength(1);
        expect(component.find('.equalSign')).toHaveLength(1);
    });

    it('renders the AC button with red class', () => {
        const component = shallow(<Button name={AC} clickHandler={jest.fn()} />);

        expect(component.find('.red')).toHaveLength(1);
    });

    it('shows no errors on console', () => {
        expect(consoleSpy).not.toHaveBeenCalled();
    });
});
