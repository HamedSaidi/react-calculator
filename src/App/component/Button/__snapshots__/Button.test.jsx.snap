// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders the Button component and matches snapshot 1`] = `
<button
  className="component-button"
  onClick={[Function]}
  type="submit"
>
  test
</button>
`;
